File: enso/engine/polyglot-api/src/main/scala/org/enso/polyglot/debugger/Response.scala
package org.enso.polyglot.debugger

import org.enso.polyglot.debugger.protocol.{
  ExceptionRepresentation,
  ObjectRepresentation
}

/** Represents a deserialized response returned from the debugger.
  *
  * It is a separate class from BinaryResponse to encapsulate Flatbuffers union
  * handling logic in a type-safe way.
  */
sealed trait Response

/** Represents an evaluation success.
  *
  * @param result over-the-wire representation of evaluation result
  */
case class EvaluationSuccess(result: ObjectRepresentation) extends Response

/** Represents an evaluation failure.
  *
  * @param exception over-the-wire representation of the exception that caused
  *                  the failure
  */
case class EvaluationFailure(exception: ExceptionRepresentation)
    extends Response

/** Represents the returned list of bindings.
  *
  * @param bindings mapping from names to bound values
  */
case class ListBindingsResult(bindings: Map[String, ObjectRepresentation])
    extends Response

/** Notification that is sent from the debugger that a REPL session should be
  * started.
  */
object SessionStartNotification extends Response

File: lib/java/ydoc-server/src/test/java/org/enso/ydoc/jsonrpc/Response.java
package org.enso.ydoc.jsonrpc;

import org.enso.ydoc.jsonrpc.result.Result;

public record Response(String jsonrpc, String id, Result result) {

  public Response(String id, Result result) {
    this("2.0", id, result);
  }

  public static Response ok(String id) {
    return new Response(id, null);
  }
}
